projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/user/entity/User.kt

```diff
@@ -218,20 +223,194 @@ data class Caseload(
     val team: Team,
 
     @Column(name = "role_code")
-    val roleCode: String
+    val roleCode: String,
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumns(
+        JoinColumn(name = "offender_id", referencedColumnName = "offender_id", insertable = false, updatable = false),
+        JoinColumn(name = "staff_employee_id", referencedColumnName = "staff_id", insertable = false, updatable = false)
+    )
+    val nextAppointment: NextAppointment? = null,
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumns(
+        JoinColumn(name = "offender_id", referencedColumnName = "offender_id", insertable = false, updatable = false),
+        JoinColumn(name = "staff_employee_id", referencedColumnName = "staff_id", insertable = false, updatable = false)
+    )
+    val previousAppointment: PreviousAppointment? = null,
+
+    @ManyToOne
+    @JoinColumn(name = "offender_id", insertable = false, updatable = false)
+    val latestSentence: LatestSentence? = null
+
+)
+
+@Entity
+@Subselect(
+    """
+        select * from(
+            select e.*,
+            row_number() over (partition by e.offender_id order by cast(e.event_number as number) desc) as row_num 
+            from event e join main_offence mo on mo.event_id = e.event_id
+            where e.soft_deleted = 0 
+            and e.active_flag = 1
+            ) sub
+        where sub.row_num = 1
+"""
+)
+@Synchronize("event")
```

Comment:
Nice - I've not really used `@Subselect` + `@Synchronize` before. Looks clever 😄

Reply:
Yeah handy for these situations